src/main/grammar/SourcePawn.bnf

```diff
@@ -3,6 +3,9 @@
   parserClass="org.idea_sp.parser.SourcePawnParser"
 
   extends="com.intellij.extapi.psi.ASTWrapperPsiElement"
+  extends(".*_expr")=expr
```

Comment:
By making all expression rules inherit from the top-level expression rule, GrammarKit apparently uses a different method of parsing (Pratt?) which eliminates left-recursion — something the regularly-scheduled parser won't handle.